capsules/src/ble_advertising_driver.rs

```diff
@@ -452,20 +455,16 @@ where
 
                 if len <= PACKET_LENGTH as u8 && result == ReturnCode::SUCCESS {
                     // write to buffer in userland
-                    let success = app
-                        .scan_buffer
-                        .as_mut()
-                        .map(|userland| {
-                            for (dst, src) in userland.iter_mut().zip(buf[0..len as usize].iter()) {
-                                *dst = *src;
-                            }
-                        })
-                        .is_some();
+                    let success = app.scan_buffer.mut_map_or(false, |userland| {
+                        for (dst, src) in userland.iter_mut().zip(buf[0..len as usize].iter()) {
```

Comment:
I think we could use https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice instead of the loop with manual copying.

`userland[0..len as usize].copy_from_slice(&buf[0..len as usize])`

Reply:
Agreed. For some edge cases, where this doesn't work (when you don't have raw access to the slice), I've verified that `.for_each(|(dst, src)| *dst = *src)` will be optimized to essentially a memcpy, but the Rust board-utils seem preferable  where applicable.